CalibTracker/SiPixelESProducers/interface/alpaka/SiPixelMappingUtilities.h

```diff
@@ -0,0 +1,57 @@
+
+#ifndef CalibTracker_SiPixelESProducers_interface_alpaka_SiPixelMappingUtilities_h
+#define CalibTracker_SiPixelESProducers_interface_alpaka_SiPixelMappingUtilities_h
+
+#include <cstdint>
+#include <alpaka/alpaka.hpp>
+#include "DataFormats/Portable/interface/alpaka/PortableCollection.h"
+#include "CalibTracker/SiPixelESProducers/interface/SiPixelMappingLayout.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixelFedCablingMap.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixelFedCablingTree.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  struct SiPixelMappingUtilities {
+    ALPAKA_FN_HOST_ACC ALPAKA_FN_ACC ALPAKA_FN_INLINE static bool hasQuality(
+        const SiPixelMappingLayoutSoAConstView& view) {
+      return view.hasQuality();
+    }
+
+    ALPAKA_FN_HOST_ACC ALPAKA_FN_ACC ALPAKA_FN_INLINE static cms::alpakatools::device_buffer<Device, unsigned char[]>
+    getModToUnpRegionalAsync(std::set<unsigned int> const& modules,
+                             const SiPixelFedCablingMap* cablingMap,
+                             Queue& queue) {
+      auto modToUnpDevice = cms::alpakatools::make_device_buffer<unsigned char[]>(queue, pixelgpudetails::MAX_SIZE);
+      auto modToUnpHost = cms::alpakatools::make_device_buffer<unsigned char[]>(queue, pixelgpudetails::MAX_SIZE);
+
+      std::vector<unsigned int> const& fedIds = cablingMap->fedIds();
+      std::unique_ptr<SiPixelFedCablingTree> const& cabling = cablingMap->cablingTree();
```

Comment:
How about passing these two from the caller to avoid constructing them for every event, since the caller (`SiPixelRawToCluster`)  already caches them by IOVs?

Reply:
Moved them in the caller.